specification/communication/data-plane/TrunkStatus/preview/2022-07-01-preview1/communicationservicestrunkstatus.json

```diff
@@ -0,0 +1,212 @@
+        "summary": "Get single Trunk health configured for resource.",
+        "operationId": "GetTrunkHealth",
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "in": "path",
+            "name": "fqdn",
+            "description": "Case insensitive FQDN of the trunk.",
+            "required": true,
+            "type": "string",
+            "maxLength": 256
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Ok.",
+            "schema": {
+              "$ref": "#/definitions/TrunkHealth"
+            }
+          },
+          "default": {
+            "description": "Failure",
+            "schema": {
+              "$ref": "../../../Common/stable/2021-03-07/common.json#/definitions/CommunicationErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Gets Trunk health.": {
+            "$ref": "./examples/GetTrunksHealth.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "ListOfTrunksHealth": {
+      "description": "Health of SIP trunks.",
+      "type": "object",
+      "properties": {
+        "value": {
+          "description": "Represents an array of SIP trunks health.",
+          "maxLength": 250,
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/TrunkHealth"
+          }
+        }
+      },
+      "required": [
+        "value"
+      ]
+    },
+    "TrunkHealth": {
+      "description": "Represents health of a SIP trunk for routing calls.",
+      "type": "object",
+      "required": [
+        "fqdn",
+        "tls",
+        "ping",
+        "overallStatus"
+      ],
+      "properties": {
+        "fqdn": {
+          "description": "Trunk's FQDN of the paired SBC",
+          "maxLength": 256,
+          "type": "string"
+        },
+        "tls": {
+          "description": "The status of the TLS connections between Direct Routing and the SBC",
+          "type": "string",
+          "enum": [
+            "NoData",
+            "OK",
+            "CertExpiring",
+            "CertExpired"
+          ],
+          "x-ms-enum": {
+            "name": "TlsStatusType",
+            "modelAsString": true
+          }
+        },
+        "ping": {
+          "description": "The status of options message send by SBC",
+          "type": "string",
+          "enum": [
+            "NoData",
+            "OK",
+            "PingError",
+            "PingExpired"
+          ],
+          "x-ms-enum": {
+            "name": "PingStatusType",
+            "modelAsString": true
+          }
+        },
+        "overallStatus": {
+          "description": "The overall status of SBC",
+          "type": "string",
+          "enum": [
+            "NoData",
+            "Active",
+            "InactiveNoRecentPings",
```

Comment:
I'm somewhat uncomfortable with enums that have combinatorial variants: `*PingsAndCalls`. How do we expect this enum to grow in the future? Do we risk blowing up the number of variants? Are clients likely to always have check for multiple related variants in their code (e.g. `if (overallStatus == 'InactiveNoRecentPings' || overallStatus == 'InactiveNoRecentCalls' || overallStatus == 'InactiveNoRecentPingsAndCalls'`)

This smells like it could be split into two separate fields: The status is limited to `NoData | Active | Inactive`. And then for inactive trunks, there are optional reasons `inactiveReasons = [](NoRecentCalls | NoRecentPings)`